src/sentry/event_manager.py

```diff
@@ -1271,48 +1346,14 @@ def save_attachments(cache_key, attachments, job):
     event = job["event"]
 
     for attachment in attachments:
-        try:
-            data = attachment.data
-        except MissingAttachmentChunks:
-            track_outcome(
-                org_id=event.project.organization_id,
-                project_id=job["project_id"],
-                key_id=job["key_id"],
-                outcome=Outcome.INVALID,
-                reason="missing_chunks",
-                timestamp=to_datetime(job["start_time"]),
-                event_id=event.event_id,
-                category=DataCategory.ATTACHMENT,
-            )
-
-            logger.exception("Missing chunks for cache_key=%s", cache_key)
-            continue
-
-        file = File.objects.create(
-            name=attachment.name,
-            type=attachment.type,
-            headers={"Content-Type": attachment.content_type},
-        )
-        file.putfile(six.BytesIO(data))
-
-        EventAttachment.objects.create(
-            event_id=event.event_id,
-            project_id=event.project_id,
-            group_id=event.group_id,
-            name=attachment.name,
-            file=file,
-        )
-
-        track_outcome(
-            org_id=event.project.organization_id,
-            project_id=job["project_id"],
+        save_attachment(
+            cache_key,
+            attachment,
+            event.project,
```

Comment:
This attribute access may hit the database. Please verify `event.project` is somehow fetched from cache first or pass `event.project_id` like the old code.

Reply:
Alternatively edit the `Event` class to fetch from cache for this property. That's probably a good idea in all cases.

Reply:
We did `event.project.organization_id` before (see [here](https://github.com/getsentry/sentry/pull/19787/files#diff-4bf345e0795c0e86cfdb6a13f60bcb64L1278)), and there are multiple code paths that already access project. In fact, the project is guaranteed to be bound at this time. 

Good catch though!